service/src/main/java/cz/muni/fi/pa165/mushrooms/service/facade/MushroomFacadeImpl.java

```diff
@@ -48,18 +48,30 @@ public MushroomDTO findMushroomByName(String name) {
 
     @Override
     public List<MushroomDTO> findByMushroomType(MushroomType mushroomType) {
+        if (mushroomType == null) {

```

Comment:
why this? Exception handling is done in DAO layer, all exceptions are wrapped in MushroomHunterServiceDataAccessException. Validity checks here go against a simple interface of the facade layer - instead of one type of exception it now throws two. (+ duplicit checks)

Reply:
When it is not there, withMock_MushroomFacadeImplTest doesn't pass this function and says: 
Expecting:
  <java.lang.NullPointerException>
to be an instance of:
  <java.lang.IllegalArgumentException>

Reply:
seems like a problem elsewhere - where?